src/Microsoft.OpenApi.OData.Reader/PathItem/MediaEntityPathItemHandler.cs

```diff
@@ -50,6 +50,15 @@ protected override void SetOperations(OpenApiPathItem item)
             {
                 AddOperation(item, OperationType.Put);
             }
+
+            DeleteRestrictionsType delete = EntitySet != null
+                ? Context.Model.GetRecord<DeleteRestrictionsType>(EntitySet)
+                : Context.Model.GetRecord<DeleteRestrictionsType>(Singleton);
+
+            if (delete == null || delete.IsDeletable)
```

Comment:
Given path is typically from a `/$value` segment, I would suggest only adding the operation if the annotation exists. This is because we may end up adding a number of paths(and increasing metadata size) that are not valid (for examples in MS graph, the deleting an message or the deleting of an attachment by the `/$value` path) which isn't useful.
This is also given that we only know of very few paths that support `delete` in the /$value segment.

Reply:
Per the OData spec, DELETE should be supported on both <path_to_entity> AND <path_to_entity>/$value [here](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_DeleteaMediaEntity).

There aren't enough media entities in the Graph metadata for the size increase to be meaningful.

Reply:
Thanks for the info!